### PR TITLE
Feat: ROR integration for institution

### PR DESCRIPTION
**Pull request by Junjiequan**
## Description
This PR aims to adapt ROR integration for Institutions. Manual ROR ID updates for already used ROR certified Institution in prod environment is required. 

Changes including :

1. Created DB patch for institutions table. Added `ROR ID` field and Dropped  `VERIFIED` field.
2. In the Institution UI, removed `verified` column and added `ROR ID` column.
3. In the Institution UI, removed `Create` button.
4. When a user login via OAuth, creates a new institution record if user's institution is not found, or updates with ROR ID otherwise ( ROR ID can be null ).

## How Has This Been Tested

- Manually

## Changes

https://github.com/UserOfficeProject/user-office-core/assets/78078898/1d334fcb-fabc-4c5a-b7a6-866931a6bc5b

## Tests included/Docs Updated?

- [ ] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: apps/frontend/src/components/institution/InstitutionTable.tsx
Comment (YufanKambang) [inline apps/frontend/src/components/institution/InstitutionTable.tsx:1]: Is this a link to an external site?

File: apps/backend/src/auth/OAuthAuthorization.ts [hunk 1]
```diff
@@ -102,17 +103,46 @@ export abstract class OAuthAuthorization extends UserAuthorization {
...
+        );
+
+    if (!institution) {
+      let institutionCountry = userInfo.institution_country
+        ? await this.adminDataSource.getCountryByName(
+            userInfo.institution_country as string
+          )
+        : null;
+
+      if (!institutionCountry || !institutionCountry.countryId) {
```
Comment (bolmsten) [inline apps/backend/src/auth/OAuthAuthorization.ts:130]: Do we need to check both institutionCountry or institutionCountry.countryId? All countries does have an id?

**Review by bolmsten — APPROVED** [review]

File: apps/frontend/src/components/institution/InstitutionTable.tsx
Comment (bolmsten) [reply apps/frontend/src/components/institution/InstitutionTable.tsx:1]: We are using ROR as a way of uniquely identifying a organisation

Comment (Junjiequan) [reply apps/frontend/src/components/institution/InstitutionTable.tsx:1]: > Is this a link to an external site?

Yes, it is external site. For example, the link will be like https://ror.org/01aj84f44

File: apps/backend/src/auth/OAuthAuthorization.ts (hunk 1 above)
Comment (Junjiequan) [reply apps/backend/src/auth/OAuthAuthorization.ts:130]: We don't need to check institutionCountry.countryId here. 
InstitutionCountry value can only be `Country | null` type. I will remove the redundant condition

**Comment by Junjiequan** [comment]
pull request from user-office-development-authorization-server needs to be merged to pass the e2e

File: apps/backend/src/datasources/UserDataSource.ts [hunk 2]
```diff
@@ -75,11 +75,7 @@ export interface UserDataSource {
   ): Promise<User>;
   ensureDummyUserExists(userId: number): Promise<User>;
   ensureDummyUsersExist(userIds: number[]): Promise<User[]>;
-  createInstitution(
-    name: string,
-    verified: boolean,
-    countryId?: number
-  ): Promise<number>;
+  createInstitution(name: string, countryId?: number): Promise<number>;
```
Comment (janosbabik) [inline apps/backend/src/datasources/UserDataSource.ts:78]: Is `rorId` not missing here?

File: apps/backend/src/datasources/postgres/records.ts [hunk 3]
```diff
@@ -353,7 +353,7 @@ export interface InstitutionRecord {
   readonly institution_id: number;
   readonly institution: string;
   readonly country_id: number;
-  readonly verified: boolean;
+  readonly ror_id?: string;
```
Comment (janosbabik) [inline apps/backend/src/datasources/postgres/records.ts:356]: Why the `ror_id` is optional? Due to the database patch, the column will be added to the Institution table, isn't it?

File: apps/backend/src/models/Institution.ts [hunk 4]
```diff
@@ -3,6 +3,6 @@ export class Institution {
     public id: number,
     public name: string,
     public country: number,
-    public verified: boolean
+    public rorId?: string
```
Comment (janosbabik) [inline apps/backend/src/models/Institution.ts:6]: Necessary optional?

**Review by janosbabik — CHANGES_REQUESTED** [review]
Looks good, just some small notes.

File: apps/backend/src/models/Institution.ts (hunk 4 above)
Comment (Junjiequan) [reply apps/backend/src/models/Institution.ts:6]: optional is necessary here, Institution can be without rorId

**Review by janosbabik — APPROVED** [review]

**Pull request merged**